src/targets/gpu/jit/reduce.cpp

```diff
@@ -131,6 +133,61 @@ static std::size_t compute_subwave_size(context& ctx, std::size_t n)
     return wavefront_size;
 }
 
+static std::vector<shape> split_reduce(const std::vector<shape>& inputs,
```

Comment:
Can you add docstring explaining what it is supposed to do ?

Reply:
Added the docstring.